### PR TITLE
new: port range supported

### PR DESCRIPTION
**Pull request by tobycroft**

### CODE REVIEW

**Comment by tobycroft** [comment]
@kuoruan

File: kcptun/kcptun.sh [hunk 1]
```diff
@@ -56,7 +56,7 @@ D_RCVWND=512
 D_DATASHARD=10
 D_PARITYSHARD=3
 D_DSCP=0
-D_NOCOMP='false'
+D_NOCOMP='true'
```
Comment (kuoruan) [inline kcptun/kcptun.sh:59]: why?

File: kcptun/kcptun.sh [hunk 2]
```diff
@@ -1201,22 +1201,37 @@ set_kcptun_config() {
...
-			echo "端口已被占用, 请重新输入!"
-			continue
-		fi
+    if [ -n "$input" ]; then
+        if echo "$input" | grep -qE '^[0-9]+-[0-9]+$'; then
+            range_start=$(echo "$input" | cut -d'-' -f1)
+            range_end=$(echo "$input" | cut -d'-' -f2)
+            if [ "$range_start" -ge 1 ] && [ "$range_start" -le 65535 ] && [ "$range_end" -ge 1 ] && [ "$range_end" -le 65535 ] && [ "$range_start" -le "$range_end" ]; then
+                if [ "$range_start" -le "$current_listen_port" ] && [ "$current_listen_port" -le "$range_end" ]; then
+                    echo "current_listen_port 在输入的范围内"
```
Comment (kuoruan) [inline kcptun/kcptun.sh:1213]: current_listen_port -> 当前监听端口

File: kcptun/kcptun.sh [hunk 3]
```diff
@@ -1201,22 +1201,56 @@ set_kcptun_config() {
...
-			[ "$listen_port" != "$current_listen_port" ]; then
-			echo "端口已被占用, 请重新输入!"
-			continue
-		fi
+    if [ -n "$input" ]; then
+        if echo "$input" | grep -qE '^[0-9]+-[0-9]+$'; then
+            range_start=$(echo "$input" | cut -d'-' -f1)
+            range_end=$(echo "$input" | cut -d'-' -f2)
+            if [ "$range_start" -ge 1 ] && [ "$range_start" -le 65535 ] && [ "$range_end" -ge 1 ] && [ "$range_end" -le 65535 ] && [ "$range_start" -le "$range_end" ]; then
+                echo "$current_listen_port"
```
Comment (kuoruan) [inline kcptun/kcptun.sh:1212]: remove this print?

**Comment by tobycroft** [comment]
@kuoruan

**Pull request merged**